LUKeychainAccess/NSKeyedUnarchiver+Additions.m

```diff
@@ -2,10 +2,12 @@
 
 @implementation NSKeyedUnarchiver (Additions)
 
-+ (nullable id)lu_unarchiveObjectOfClass:(Class)cls withData:(NSData *)data {
++ (nullable id)lu_unarchiveObjectOfClasses:(NSSet*)set withData:(NSData *)data {
   if (@available(iOS 11, *)) {
     NSError *error;
-    id result = [self unarchivedObjectOfClass:cls fromData:data error:&error];
+    NSSet *_set = [NSSet setWithArray:@[NSString.class, NSDictionary.class, NSArray.class, NSNumber.class]];
```

Comment:
This is a tough call to make on exactly what classes we should be supporting for all of our unarchiving needs. 

I decided on the main collection and data types but an argument could me made to include NSDate.class, NSURL.class since those are pretty common data types as well, but then do we also include NSUUID.class, NSNull.class and I'm sure there are others that could be added that I'm not thinking of. I just didn't want to have to put an update out every time we noticed a missing NS class that should be in the "defaultSet".

Another idea is to potentially not have a default `_set` at all and require each call to include these base classes in their passed in Set. Thoughts

Reply:
I think the fact that your inspiration comes from Google is a good sign. Is there a way we can test to see whether this fails whenever we attempt to unarchive something from the keychain that is not included by `_set`? If it works without them included by `_set`, what's the motivation for adding them?

Reply:
I added a test to confirm if you don't pass in the correct classes an error is thrown. Bascially NSString, NSNumber NSArray and NSDictionary are already passed in as "allowed classes" for the unarchiver which is what the Google PR is doing as well. After discussing with @Xeaza I think we're going to keep it to just this list. This just means for any other Objective C class type (NSUUID, NSDate, etc.)  along with our own custom types, have to be explicitly mentioned when calling this method.